.github/workflows/portal-loop.yml

```diff
@@ -46,3 +48,57 @@ jobs:
           push: ${{ github.event_name != 'pull_request' }}
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
+
+  test-portal-loop-docker-compose:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: "Checkout"
+        uses: actions/checkout@v4
+
+      - name: "Setup the images"
+        run: |
+          cd misc/loop
+
+          docker compose build
+          docker compose pull
+          docker compose up -d
+
+      - name: "Test1 - Portal loop start gnoland"
+        run: |
+          while
+            block_height=$(curl -s localhost:26657/status | jq -r '.result.sync_info.latest_block_height')
+            echo "Current block height: $block_height"
+            [[ "$block_height" -lt 10 ]]
+          do
+            sleep 1
+          done
+
+          curl -s localhost:26657/status | jq
+
+      - name: "Buid new gnolang/gno image"
+        run: |
+          docker build -t ghcr.io/gnolang/gno -f Dockerfile.gnoland.release .
+
+      - name: "Wait for new docker image"
+        run: |
+          ip_addr=$(cat misc/loop/traefik/gno.yml | grep -o "http://.*:26657")
+          while
+            new_ip_addr=$(cat misc/loop/traefik/gno.yml | grep -o "http://.*:26657")
+            echo "${ip_addr} -> ${new_ip_addr}"
+            [[ "${ip_addr}" == ${new_ip_addr} ]]
+          do
+            sleep 5
+          done
```

Comment:
where is gno.yml file modified? why should we do this instead of waiting for the command that is creating the docker image? this might finish in a timeout if something goes wrong.

Reply:
This is how the portal loop works, once the "switch" is done, it write in the `gno.yml` file (which is traefik config) the new url.

i put a timeout of 10minutes for the workflow, if it's timeout, it's a fail.